`coder external-auth access-token` panic when no provider name is given

```console
coder@test:~$ coder external-auth access-token
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/coder/coder/v2/cli.(*RootCmd).externalAuthAccessToken.func1(0xc00068c200)
        /home/runner/actions-runner/_work/coder/coder/cli/externalauth.go:75 +0x778
github.com/coder/coder/v2/cli.(*RootCmd).Command.func6.1(0x0?)
        /home/runner/actions-runner/_work/coder/coder/cli/root.go:282 +0xd1
github.com/coder/coder/v2/cli/clibase.(*Invocation).run(0xc00068c200, 0xc00059be00)
        /home/runner/actions-runner/_work/coder/coder/cli/clibase/cmd.go:388 +0x10b2
github.com/coder/coder/v2/cli/clibase.(*Invocation).run(0xc00068c000, 0xc00059be00)
        /home/runner/actions-runner/_work/coder/coder/cli/clibase/cmd.go:325 +0xb5e
github.com/coder/coder/v2/cli/clibase.(*Invocation).run(0xc00068c000, 0xc00059be00)
        /home/runner/actions-runner/_work/coder/coder/cli/clibase/cmd.go:325 +0xb5e
github.com/coder/coder/v2/cli/clibase.(*Invocation).Run(0xc00059beb8?)
        /home/runner/actions-runner/_work/coder/coder/cli/clibase/cmd.go:478 +0xe5
github.com/coder/coder/v2/cli.(*RootCmd).RunMain(0xc000225400, {0xc000626000, 0x26, 0x40})
        /home/runner/actions-runner/_work/coder/coder/cli/root.go:137 +0x1aa
main.main()
        /home/runner/actions-runner/_work/coder/coder/enterprise/cmd/coder/main.go:11 +0x45
 ```

--- a/cli/externalauth.go
+++ b/cli/externalauth.go
@@ -52,6 +52,9 @@ fi
 				Command:     "coder external-auth access-token slack --extra \"authed_user.id\"",
 			},
 		),
+		Middleware: clibase.Chain(
+			clibase.RequireNArgs(1),
+		),
 		Options: clibase.OptionSet{{
 			Name:        "Extra",
 			Flag:        "extra",

--- a/cli/externalauth_test.go
+++ b/cli/externalauth_test.go
@@ -46,6 +46,19 @@ func TestExternalAuth(t *testing.T) {
 		clitest.Start(t, inv)
 		pty.ExpectMatch("bananas")
 	})
+	t.Run("NoArgs", func(t *testing.T) {
+		t.Parallel()
+		srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			httpapi.Write(context.Background(), w, http.StatusOK, agentsdk.ExternalAuthResponse{
+				AccessToken: "bananas",
+			})
+		}))
+		t.Cleanup(srv.Close)
+		url := srv.URL
+		inv, _ := clitest.New(t, "--agent-url", url, "external-auth", "access-token")
+		watier := clitest.StartWithWaiter(t, inv)
+		watier.RequireContains("wanted 1 args but got 0")
+	})
 	t.Run("SuccessWithExtra", func(t *testing.T) {
 		t.Parallel()
 		srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {

